buildSrc/src/main/java/saros/gradle/intellij/ConfigureIntelliJTask.java

```diff
@@ -0,0 +1,121 @@
+import java.io.File;
+import java.io.IOException;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.intellij.IntelliJPluginExtension;
+
+public class ConfigureIntelliJTask extends DefaultTask {
+
+  private static final String LOCKFILE_NAME = "saros_sandbox.lock";
+  private static final String SANDBOX_DIR_PREFIX = "idea-sandbox-";
+
+  public ConfigureIntelliJTask() {
+    super();
+  }
+
+  @TaskAction
+  public void configure() {
+    Project project = getProject();
+    IntelliJPluginExtension intellijExtension =
+        project.getExtensions().findByType(IntelliJPluginExtension.class);
+    SarosIntellijExtension extension =
+        project.getExtensions().findByType(SarosIntellijExtension.class);
+
+    configureDefaultParams(intellijExtension, extension);
+    configureRunIdeIntellijInstallation(intellijExtension, extension, project);
+  }
+
+  /**
+   * Configures the intellij plugin, but also adds actions to the task runIde (provided by the
+   * intellij plugin) that lock the running intellij instance and use another intellij config dir if
+   * the current dir is locked. This allows to run multiple intellij instances with runIde
+   * simultaneously.
+   */
+  private void configureRunIdeIntellijInstallation(
+      IntelliJPluginExtension intellijExtension,
+      SarosIntellijExtension extension,
+      Project project) {
+
+    // Set intellij installation which is used by runIde
+    File localIntellijHome = extension.getLocalIntellijHome();
+    if (localIntellijHome != null
+        && localIntellijHome.isDirectory()
+        && localIntellijHome.exists()) {
+      intellijExtension.setLocalPath(localIntellijHome.getAbsolutePath());
+    } else {
+      intellijExtension.setVersion(extension.getIntellijVersion());
+    }
+
+    // Determine sandbox directory which contains the intellij configurations and logs
+    File sandboxDir = determineSandboxDir(extension, project);
+    intellijExtension.setSandboxDirectory(sandboxDir.getAbsolutePath());
+
+    configureLockFileActions(sandboxDir, project);
+  }
+
+  /** Sets the default parameters of the intellij plugin */
+  private void configureDefaultParams(
+      IntelliJPluginExtension intellijExtension, SarosIntellijExtension extension) {
+    intellijExtension.setUpdateSinceUntilBuild(false);
+    // TODO: Use more generic so
+    intellijExtension.setPluginName(extension.getPluginName());
+  }
+
+  /**
+   * Add tasks that locks the runIde sandbox when the runIde task is started and unlocks the sandbox
+   * when runIde is finished.
+   */
+  private void configureLockFileActions(File sandboxDir, Project project) {
+    Task runIdeTask = project.getTasks().findByPath("runIde");
+    runIdeTask.doFirst(
+        (Task t) -> {
+          try {
+            sandboxDir.mkdirs();
+            getLockFile(sandboxDir).createNewFile();
```

Comment:
Check whether file was successfully created? I haven't had a look at the logic ensuring different sandboxes are used yet.